docs/performance-tests.md

```diff
@@ -0,0 +1,218 @@
+
+Mezo doesn't implement this dynamic block sizing capability. Instead, Mezo maintains fixed
+block gas limits, which means its capacity cannot expand during high demand periods.
+
+This design choice prioritize predictable block production times (expected to be 3.5 seconds),
+but it does sacrifice some of the flexibility that Ethereum's implementation offers during
+usage spikes.
+
+#### Ethereum vs Mezo mempools behaviour
+
+A critical architectural difference between Ethereum and Mezo lies in their mempool
+management strategies. In Ethereum's implementation, transactions can remain in the mempool
+indefinitely, regardless of changing network conditions or base fee fluctuations. This persistence
+allows transactions to eventually be included when network congestion subsides or when
+their gas price becomes competitive again relative to current demand.
+
+Mezo, however, employs a more aggressive pruning approach to mempool management. When the
+minimum gas fee for block inclusion increases due to network congestion, Mezo actively
+removes transactions from the mempool whose gas price falls below this threshold. This is
+fundamentally different from Ethereum's model, as it means transactions can be rejected
+from the network entirely rather than simply waiting for inclusion during periods of lower demand.
+
+This behavior is enforced through CometBFT (re)checkTx mecanism which continuously re-evaluates
+all transactions in the mempool against current network conditions. While this approach
+keeps the mempool cleaner and more reflective of current fee market conditions, it introduces
+a significant computational overhead. As the mempool backlog grows, the constant validation
+and re-checking of all pending transactions can consume substantial node resources, potentially
+affecting overall network performance during sustained high-demand periods
+
+For users, this distinction means that transaction submission strategies must be more
+carefully thought on Mezo than on Ethereum. Transactions submitted with marginally
+competitive gas prices that might eventually clear on Ethereum could be completely removed
+from consideration on Mezo, requiring resubmission with higher fees.
+
+#### Gas Limit per transaction and block sizing
+
+Mezo takes a different approach to determining which transactions fit into a block compared
+to Ethereum, particularly regarding block size calculations and gas consumption mechanics.
+
+In Mezo, the effective space in a block is determined by a fixed gas limit per block, similar
+to Ethereum. However, Mezo's transaction selection process differs significantly in how
+it accounts for gas consumption:
+
+##### Gas Accounting Differences
+
+While Ethereum charges based on the actual computational resources used during execution (with
+an upfront maximum specified by the user's gasLimit), Mezo implements a minimum gas consumption
+rule. Specifically, Mezo will consume at least half of the gasLimit specified in a transaction,
```

Comment:
This parameter is configurable. 50% is just the default value. Worth mentioning it.

Reply:
fair point. I'll add a line about it

Reply:
addressed in 90cbdc8ea7b82c777fbc258fd2d03d5c96631b75